src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -87,130 +96,142 @@ public function handle(AddProductToOrderCommand $command)
     {
         $order = $this->getOrderObject($command->getOrderId());
 
-        if ($this->context->cart !== null) {
-            $oldCartRules = $this->context->cart->getCartRules();
-        } else {
-            $oldCartRules = [];
-        }
+        $this->contextStateManager
+            ->setCurrency(new Currency($order->id_currency))
+        ;
 
-        $this->assertOrderWasNotShipped($order);
+        try {
+            if ($this->context->cart !== null) {
```

Comment:
I feels weird to use the context cart, I guess it comes from the legacy controller

Isn't it possible to get the cart from the order instead? And it could be overridden in the context like the Currency is with the `contextStateManager`

Reply:
Done

Reply:
Done